be/src/vec/exec/join/vhash_join_node.cpp

```diff
@@ -317,10 +319,10 @@ struct ProcessHashTableProbe {
                 for (auto it = mapped.begin(); it.ok(); ++it) {
                     ++current_offset;
                     for (size_t j = 0; j < right_col_len; ++j) {
-                        auto& column = *it->block->get_by_position(j).column;
+                        auto& column = *it->block()->get_by_position(j).column;
                         mcol[j + right_col_idx]->insert_from(column, it->row_num);
                     }
-                    visited_map.emplace_back(&it->visited);
+                    visited_map.emplace_back(&(*it));
```

Comment:
why not just `emplace_back(it)`?

Reply:
&(*it) is not same to it, this is because ForwardIterator has overloaded the operator*